src/main/java/com/griddynamics/cd/nrp/internal/uploading/impl/factories/JerseyClientFactory.java

```diff
@@ -0,0 +1,61 @@
+package com.griddynamics.cd.nrp.internal.uploading.impl.factories;
+
+import com.griddynamics.cd.nrp.internal.model.config.ReplicationPluginConfigurationStorage;
+import com.sun.jersey.api.client.Client;
+import com.sun.jersey.api.client.config.ClientConfig;
+import com.sun.jersey.api.client.config.DefaultClientConfig;
+import com.sun.jersey.api.client.filter.HTTPBasicAuthFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+@Singleton
+public class JerseyClientFactory {
+
+    private Logger log = LoggerFactory.getLogger(JerseyClientFactory.class);
+
+    private ExecutorService executorService;
+    private final ReplicationPluginConfigurationStorage
+            replicationPluginConfigurationStorage;
+
+    @Inject
+    public JerseyClientFactory(
+            ReplicationPluginConfigurationStorage
+                    replicationPluginConfigurationStorage ) {
+        this.replicationPluginConfigurationStorage = replicationPluginConfigurationStorage;
+
+    }
+
+    public void onActivate(){
+        int requestQueueSize = replicationPluginConfigurationStorage.getRequestQueueSize();
+        this.executorService = new ThreadPoolExecutor(
+                replicationPluginConfigurationStorage.getRequestSendingThreadCount(),
+                replicationPluginConfigurationStorage.getRequestSendingThreadCount(),
+                30,
+                TimeUnit.SECONDS,
+                new LinkedBlockingQueue<Runnable>(
+                        requestQueueSize));
+    }
+
+    public Client getClient(String login, String password) {
+        ClientConfig config = new DefaultClientConfig();
+        config.getProperties().put(ClientConfig.PROPERTY_CONNECT_TIMEOUT, 1000);
+        config.getProperties().put(ClientConfig.PROPERTY_READ_TIMEOUT, 2000);
```

Comment:
Could you also leave comments on why you set those values and why particularly _these_ values?